### PR TITLE
Add Python anagram checker

### PR DESCRIPTION
**Pull request by andrey-unblocked**
python/anagram.py — checks if two strings are anagrams.

### CODE REVIEW

**Review by andrey-unblocked — COMMENTED** [review]
The PR introduces a new pure function `is_anagram` with no test coverage. Per team best practices, every new pure function should be accompanied by parametrized tests covering boundary cases (empty strings, single characters, mixed case, spaces-only, large values) to prevent silent regressions on future refactors.

File: python/anagram.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    sorted_a = sorted(a)
+    # sort the letters of b
+    sorted_b = sorted(b)
+
+    # compare the two sorted lists - this uses bubble sort under the hood
+    # (Python's sorted() actually uses Timsort, not bubble sort, but the
+    # effect is the same for our purposes here)
+    # return True if they match
+    if sorted_a == sorted_b:
+        return True
```
Comment (andrey-unblocked) [inline python/anagram.py:33]: **Write parametrized tests covering boundary cases for every new pure function.**

`is_anagram` is a pure function with no accompanying test file. No boundary cases are covered (empty strings, single characters, strings with only spaces, case variations, large inputs).

Ref prior reviewer guidance: https://github.com/Andrey-Test-Org/kittycat/pull/27#discussion_r3305620769 — "No tests cover this function. Boundary cases ... are the spots that quietly regress on refactor — a short test file with `pytest.mark.parametrize` would lock the contract."
